balancer.go

```diff
@@ -83,11 +136,99 @@ func (info topicInfo) RoundRobin() map[string][]int32 {
 	return res
 }
 
+func rebalanceWithStrategy(topics map[string]TopicInfo,
+	rebalanceFunc func(TopicInfo) map[string][]int32,
+) map[string]map[string][]int32 {
+
+	res := make(map[string]map[string][]int32, 1)
+	for topic, info := range topics {
+		for memberID, partitions := range rebalanceFunc(info) {
+			if _, ok := res[memberID]; !ok {
+				res[memberID] = make(map[string][]int32, 1)
+			}
+			res[memberID][topic] = partitions
+		}
+	}
+	return res
+}
+
+// --------------------------------------------------------------------
+
+// StripedRebalancer implements StrategyStriped, balancing topics fairly
+// across consumers
+type StripedBalancer struct{}
+
+// AllMembersForGroup the unique set of group members for the given group info map.
+func AllMembersForGroup(group map[string]TopicInfo) []string {
+	memberMap := make(map[string]bool)
+
+	for _, topic := range group {
+		for _, memberID := range topic.MemberIDs {
+			memberMap[memberID] = true
+		}
+	}
+
+	members := make([]string, 0, len(memberMap))
+	for memberID, _ := range memberMap {
+		members = append(members, memberID)
+	}
+	return members
+}
+
+// TopicsAndPartitionsForGroup returns a map of topics with a slice containings
+// their partitions.
+func TopicsAndPartitionsForGroup(group map[string]TopicInfo) map[string][]int32 {
+	topicMap := make(map[string][]int32)
+	for topic, info := range group {
+		topicMap[topic] = make([]int32, 0, len(info.Partitions))
+		for _, partition := range info.Partitions {
+			topicMap[topic] = append(topicMap[topic], partition)
+		}
+	}
+	return topicMap
+}
+
+func sortedMapKeys(m map[string][]int32) []string {
+	ret := make([]string, 0, len(m))
+	for key, _ := range m {
+		ret = append(ret, key)
+	}
+
+	sort.Strings(ret)
+	return ret
+}
+
+// Rebalance rebalances the consumer group using the striped strategy.
+func (sb *StripedBalancer) Rebalance(topics map[string]TopicInfo) map[string]map[string][]int32 {
+	memberIDs := AllMembersForGroup(topics)
```

Comment:
So yes, `topicInfoGroup` would simplify things here and can be unit-tested individually